agent/provider/src/cli/rule.rs

```diff
@@ -70,7 +84,36 @@ fn set(set_rule: SetRule, config: ProviderConfig) -> Result<()> {
                 Some(_) => todo!("Setting rule for specific certificate isn't implemented yet"),
                 None => rules.set_default_audited_payload_mode(mode),
             },
-            SetOutboundRule::Partner { cert_id, mode } => rules.set_partner_mode(cert_id, mode),
+            SetOutboundRule::Partner(RuleWithCert::CertId { cert_id, mode }) => {
+                rules.set_partner_mode(cert_id, mode)
+            }
+            SetOutboundRule::Partner(RuleWithCert::ImportCert { import_cert, mode }) => {
+                let keystore_manager = KeystoreManager::try_new(&rules.cert_dir)?;
+
+                let KeystoreLoadResult { loaded, skipped } =
+                    keystore_manager.load_certs(&vec![import_cert])?;
+
+                //TODO it will be removed after backward compatibility is done
```

Comment:
I wouldn't like to see this forgotten, can you create a task for that and mark it as blocked by backwards compat?

Reply:
Done in https://github.com/golemfactory/yagna/issues/2445